mount-partitions.sh

```diff
@@ -30,14 +30,58 @@ dbus_get_mount() {
     fi
 }
 
+# Identify an encrypted partition using dmsetup
+# Works for both dm-crypt and LUKS encrypted partitions
+# Arguments:
+# 1: Partition device - will be converted to a DM device
+# Returns:
+# 0: The partition device is encrypted
+# 1: The partition device is not encrypted
+is_part_encrypted() {
+	_part="${1#/dev/}"
+	_dm_part="${_part}"
+	if command -v dmsetup > /dev/null; then
+		if [ "${_part#dm-}" = "${_part}" ]; then
+			# Does not start with dm-
+			if [ "${_part#mapper/}" = "${_part}" ]; then
+				# Does not start with mapper/
```

Comment:
```suggestion
		if [ "${_part#dm-}" = "${_part}" ] || [ "${_part#mapper/}" = "${_part}" ]; then
			# does not start with `dm-` or `mapper/`
```

Reply:
@pipex I think you meant to use `&&`:
```
if [ "${_part#dm-}" = "${_part}" ] && [ "${_part#mapper/}" = "${_part}" ]
```
I preferred the two lines as it is more legible and the `&&` is implicit.